spark/src/main/scala/org/apache/spark/sql/delta/DeltaHistoryManager.scala

```diff
@@ -67,46 +67,48 @@ class DeltaHistoryManager(
    * chronological order.
    */
   def getHistory(limitOpt: Option[Int]): Seq[DeltaHistory] = {
+    val snapshot = deltaLog.update()
     val listStart = limitOpt.map { limit =>
-      math.max(deltaLog.update().version - limit + 1, 0)
+      math.max(snapshot.version - limit + 1, 0)
     }.getOrElse(getEarliestDeltaFile(deltaLog))
-    getHistory(listStart)
+    getHistory(listStart, end = Some(snapshot.version))
   }
 
   /**
-   * Get the commit information of the Delta table from commit `[start, end]`. If `end` is `None`,
-   * we return all commits from start to now.
+   * Get the commit information of the Delta table from commit `[start, end)`.
+   *
+   * @param isRangeICT Whether the commit range is ICT-enabled.
+   *                   If `true`, all commits in the range must have ICTs and
+   *                   the timestamp returned for each commit will be the ICT.
+   *                   If `false`, the file modification time will be used as the timestamp.
    */
-  def getHistory(
+  private[delta] def getHistoryImpl(
```

Comment:
Can we make this `end` inclusive as well (and upperBoundExclusive -> upperBoundInclusive) for consistency across different sites in this file?

Reply:
Done